fix(docs-infra): amend color of code links inside a tags

dario-piotrowicz opened the pull request:
when some auto code links fail to happen they are added manually
with the md ``[`code text`](link)``, these generate anchor elements which
contain a code element, such code element does not get the correct text
color, this commit fixes such issue

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] docs style change


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A

Code links automatically generated by the docs pre-processor get a light blue color, but those added manually in md files don't

For example:
![Screenshot at 2021-09-25 13-12-39](https://user-images.githubusercontent.com/61631103/134771182-9e5f2e0e-7424-4627-ac05-3f0d2e4d7008.png)


## What is the new behavior?

Code links manually added in md files get the blue color as the other auto-generated links:

For example:
![Screenshot at 2021-09-25 13-13-02](https://user-images.githubusercontent.com/61631103/134771229-57141aa4-297a-4e4d-9ec4-e8081f68988a.png)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

dario-piotrowicz commented:
### Note
Unfortunately this does not make the manual links 100% the same as auto-generates as in functions the brackets gets also colored/included in the link, which is not what happens with the auto-generated links:
![Screenshot at 2021-09-25 13-34-25](https://user-images.githubusercontent.com/61631103/134771924-7a9c8af7-012d-46a5-84d7-6ed6b9ff42b7.png) vs
![Screenshot at 2021-09-25 13-37-14](https://user-images.githubusercontent.com/61631103/134771926-857a6053-c27a-4437-963d-c23bd8dbea45.png)

This is unfortunate but I don't think there is much that can be done here about it as that is related to the docs generation

petebacondarwin reviewed (approved):
Thanks @dario-piotrowicz.

I agree that this does not make things worse. And in the bigger picture, I am not worried about the difference TBH.

By the way, you can use double backticks to mark inline code that will contain backticks to make your git message cleaner:

```
``[`code text`](link)``
```

dario-piotrowicz commented:
> Thanks @dario-piotrowicz.
> 
> I agree that this does not make things worse. And in the bigger picture, I am not worried about the difference TBH.
> 
> By the way, you can use double backticks to mark inline code that will contain backticks to make your git message cleaner:
> 
> ```
> ``[`code text`](link)``
> ```

Thanks, I've updated the commit message with the double backticks :slightly_smiling_face::+1: 

(about the difference with the brackets... yeah it's a bit of an unimportant detail, I am sure there are much more relevant things than that :sweat_smile:)

gkalpak commented on aio/src/styles/2-modules/code/_code-theme.scss:
```diff
@@ -87,17 +87,22 @@
 
     :not(h1):not(h2):not(h3):not(h4):not(h5):not(h6):not(pre) {
-      > code {
+      & > code {
```
OOC, why this change?

dario-piotrowicz replied:
... personal preference :see_no_evil: :sweat_smile: :disappointed:

dario-piotrowicz replied:
is this ok? :sweat_smile: :sweat: :see_no_evil:

gkalpak replied:
I am not a big fan (and it is generally better to remain consistent), but I am not going to block this PR over this :wink:

dario-piotrowicz replied:
I imagined that `& >` was actually more consistent (I didn't really look into it :sweat:)

if you prefer the other version I will just revert it sorry :slightly_smiling_face::+1:

gkalpak commented on aio/src/styles/2-modules/code/_code-theme.scss:
```diff
@@ -87,17 +87,22 @@
         color: if($is-dark-theme, constants.$white, constants.$darkgray);
       }
+
+      &:is(a) > code {
```
I would rather not use `:is()` yet (at least until we switch to v13 and drop IE - and also Safari has is not 100% spec-compliant).

How about:

```diff
   .sidenav-content {
     code {
       a {
         color: if($is-dark-theme, constants.$cyan, constants.$darkblue);
       }
     }
 
     :not(h1):not(h2):not(h3):not(h4):not(h5):not(h6):not(pre) {
       > code {
         background-color: if($is-dark-theme, constants.$darkgray, rgba(constants.$lightgray, 0.3));
-        color: if($is-dark-theme, constants.$white, constants.$darkgray);
+
+        &:not(a) {
+          color: if($is-dark-theme, constants.$white, constants.$darkgray);
+        }
       }
     }
   }
```

dario-piotrowicz replied:
yes totally agree with avoiding the `:is` I did it on my other PR but forgot to do it here :sweat: 

yeah I think your change should definitely be the right solution, I will test it out this evening and commit it :slightly_smiling_face::+1:

Thanks a lot :smiley: 

(and sorry yeah....my solution was pretty ugly wasn't it :sweat_smile:)

dario-piotrowicz replied:
Hi @gkalpak sorry I could not go with your solution as it didn't work :sweat: 

the issue is that your `&:not(a)` isn't what we need, `a` is the parent and `code` is the element we want to target :sweat: 

So I went with this: https://github.com/angular/angular/pull/43586/commits/3ff1f3fa2bca9bf528bfbddccf6dd37f81872cf7

Please have a look it's not as clean but it removes both the duplication and the `:is` :slightly_smiling_face:

gkalpak replied:
Oh, you're right 🤦‍♂️
Yeah, I meant, `$:not(a) > code` (you got the spirit, though 😛).

gkalpak reviewed (changes requested):
Thx, @dario-piotrowicz ❤️
One suggestion to avoid `:is()` and also avoid style duplication. Otherwise lgtm ✨

gkalpak commented on aio/src/styles/2-modules/code/_code-theme.scss:
```diff
@@ -87,21 +87,17 @@
-
-      &:is(a) > code {
-        color: $link-color;
+      &:not(a) > code {
```
```suggestion

      &:not(a) > code {
```

😱

dario-piotrowicz replied:
sorry :sweat:

mary-poppins commented:
You can preview ffeaf3a at https://pr43586-ffeaf3a.ngbuilds.io/.
You can preview 680dd1b at https://pr43586-680dd1b.ngbuilds.io/.
You can preview 3ff1f3f at https://pr43586-3ff1f3f.ngbuilds.io/.

gkalpak reviewed (approved):
Thx, @dario-piotrowicz! (One minor nit - otherwise good to go 🚀)

gkalpak commented:
Thx, @dario-piotrowicz! This should be good to go as soon as CI is green. (It seems that a rebase is needed currently.)

dario-piotrowicz commented:
> Thx, @dario-piotrowicz! This should be good to go as soon as CI is green. (It seems that a rebase is needed currently.)

Actually this should be up to date already :confused:

gkalpak commented:
> Actually this should be up to date already 😕

Well, CI [strongly disagrees][1] :grin:
I'll rebase this.

[1]: https://circleci.com/gh/angular/angular/1061241

dario-piotrowicz committed:
fix(docs-infra): amend color of code links inside single anchors

when some auto code links fail to happen they can be added manually
with the md ``[`code text`](link)``, these generate anchor elements which
contain a code element, such code element does not get the correct text
color, this commit fixes such issue

dario-piotrowicz committed:
fixup! fix(docs-infra): amend color of code links inside single anchors

dario-piotrowicz committed:
fixup! fix(docs-infra): amend color of code links inside single anchors

mary-poppins commented:
You can preview f22f474 at https://pr43586-f22f474.ngbuilds.io/.

dario-piotrowicz commented:
> > Actually this should be up to date already confused
> 
> Well, CI [strongly disagrees](https://circleci.com/gh/angular/angular/1061241) grin I'll rebase this.

Sorry.... I did check it and the branch seemed up to date.... sorry :sweat:

gkalpak commented:
No worries :smiley:

dylhunn commented:
This PR was merged into the repository by commit 2a58ff3e416e2cfdec0e670489f9ebe5baaf2e22.
